egui/src/memory.rs

```diff
@@ -373,6 +374,44 @@ impl Memory {
     }
 }
 
+impl Memory {
+    pub fn get<T: DataElementTrait>(&mut self, id: Id) -> Option<&T> {
+        self.data.get_mut(&id)?.get()
+    }
+
+    pub fn get_or_insert<T: DataElementTrait>(&mut self, id: Id, or_insert: T) -> Option<&T> {
+        self.data
+            .entry(id)
+            .or_insert_with(|| DataElement::new(or_insert))
+            .get()
+    }
+
+    pub fn set<T: DataElementTrait>(&mut self, id: Id, set: T) {
+        self.data.insert(id, DataElement::new(set));
+    }
+
+    // Not all can be resetted, if it's not deserialized
+    pub fn reset_all<T: DataElementTrait>(&mut self) {
+        self.data
+            .retain(|_, v| (*v).type_id() == Some(std::any::TypeId::of::<T>()));
+    }
+
+    // Not anything can be counted
+    pub fn count<T: DataElementTrait>(&mut self) -> usize {
+        self.data
+            .iter()
+            .filter(|(_, v)| (*v).type_id() == Some(std::any::TypeId::of::<T>()))
+            .count()
+    }
+
+    pub fn count_unknown(&mut self) -> usize {
+        self.data
+            .iter()
+            .filter(|(_, v)| (*v).type_id().is_some())
+            .count()
+    }
+}
```

Comment:
I think it would be nice if this interface was put on a newtype, e.g.

`pub struct DataStore(HashMap<Id, DataElement>)`

Reply:
Done https://github.com/emilk/egui/pull/257/commits/81ff915cd127bc7864dde58b416a0189b89a65eb